themes/default/content/docs/using-pulumi/pulumi-packages/debugging-provider-packages.md

```diff
@@ -67,3 +67,49 @@ Once Pulumi runs or tests are initiated with the `PULUMI_DEBUG_PROVIDERS` enviro
 {{< notes type="warning" >}}
 **Terminating the Provider Process**: Be cautious when terminating the provider process as the Pulumi state can get out of sync with the actual cloud resources. When in doubt, `pulumi refresh` will address this.
 {{< /notes >}}
+
+## Debug tfgen
+
+If you need to debug tfgen in one of the providers, you first have to start
+tfgen using [dlv](https://github.com/go-delve/delve) exec.
+
+### Install dlv
+
+```shell
+go install github.com/go-delve/delve/cmd/dlv@latest
+```
+
+### Run tfgen with dlv exec
+
+```shell
+make debug_tfgen
+```
+
+### Using local pulumi-terraform-bridge
+
+If you want to use a local `pulumi-terraform-bridge` you can use [go
+workspaces](https://go.dev/doc/tutorial/workspaces) to build the local provider
+along with the local `pulumi-terraform-bridge`.
+
+```shell
+go work init
+go work use -r ./provider
+go work use -r ../pulumi-terraform-bridge
+```
```

Comment:
We should make it clear what the directory structure here is.

I would add something like this, but I don't have strong opinions on how we show this:

```shell
$ ls
pulumi-my-provider        # YOUR provider
pulumi-terraform-bridge   # github.com/pulumi/pulumi-terraform-bridge

$ cd pulumi-my-provider
```

Reply:
Updated!